<launch>
	<arg name="sigverse_ros_bridge_port"        default="50001" />
	<arg name="ros_bridge_port"                 default="9090" />

	<arg name="camera"  default="camera" />

	<group ns="recognize_pointed_direction">
		<node name="recognize_pointed_direction" pkg="sigverse_turtlebot3_open_manipulator" type="recognize_pointed_direction" output="screen">
		<!--
		<remap from="/tb3omc/instruction"          to="/tb3omc/instruction" />
		<remap from="/camera/depth/image_raw"      to="/camera/depth/image_raw" />
		<remap from="/tb3omc/joint_trajectory"     to="/tb3omc/joint_trajectory" />
		-->
		</node>
	</group>

	<group ns="sigverse_ros_bridge">
		<node name="sigverse_ros_bridge" pkg="sigverse_ros_bridge" type="sigverse_ros_bridge" args="$(arg sigverse_ros_bridge_port)"/>
	</group>
	
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" > 
		<arg name="port" value="$(arg ros_bridge_port)"/>
	</include>

	<node pkg="image_view" type="image_view" name="image_view" args="image:=$(arg camera)/depth/image_raw"/>

</launch>

